packages/cli/src/workflows/workflows.controller.ts

```diff
@@ -62,6 +66,24 @@ workflowsController.post(
 
 		WorkflowHelpers.addNodeIds(newWorkflow);
 
+		if (isSharingEnabled()) {
```

Comment:
I kinda liked that we had two handlers for the same route, and we routed the requested correctly via a middleware. 
We don't need to change this now, but once we convert this router into a decorated class, I'd like to split these back up again into separate methods.

Reply:
Yes, definitely agree. This is an interim state.